source/MaterialXContrib/MaterialXNode/scripts/MaterialXNodeUtil.py

```diff
@@ -0,0 +1,35 @@
+import os
+import maya.cmds as cmds
+import maya.mel as mel
+
+from subprocess import Popen
+from distutils.spawn import find_executable
+
+def editMaterialXDocument(documentPath, element, editor):
+    editor_path = find_executable(editor)
+    if editor_path == None:
+        print ("Unable to find editor: " + editor)
+    else:
+        print "Launching editor..."
+        cmd = "system(\"start " + editor + " \\\"" + documentPath + "\\\"\");"
+        mel.eval(cmd)
+
+def editMaterialXNode(nodeName):
+    documentFilePath = cmds.getAttr(nodeName + "documentFilePath")
+    elementPath = cmds.getAttr(nodeName + "elementPath")
+    editor = os.environ['MAYA_MATERIALX_EDITOR'] if 'MAYA_MATERIALX_EDITOR' in os.environ else "Notepad.exe"
```

Comment:
There is no notepad on Linux or Mac so maybe make this vi on those platforms.

Reply:
I think there is actually a Maya text editor preference but maybe it's easiest to make this an optionVar.